packages/core/admin/ee/server/register.js

```diff
@@ -4,8 +4,31 @@ const { features } = require('@strapi/strapi/lib/utils/ee');
 const executeCERegister = require('../../server/register');
 const createAuditLogsService = require('./services/audit-logs');
 
+// Migrate the audit logs table name for users coming from v4.6.0
+const migrateAuditLogsTable = async ({ oldContentTypes, contentTypes }) => {
+  // Check if the table name needs to be migrated
+  const oldName = oldContentTypes?.['admin::audit-log']?.collectionName;
+  const newName = contentTypes['admin::audit-log']?.collectionName;
+  const shouldMigrate = oldName === 'audit_logs' && newName === 'strapi_audit_logs';
+
+  if (shouldMigrate) {
+    // Migrate the main audit logs table
+    if (await strapi.db.getSchemaConnection().hasTable('audit_logs')) {
```

Comment:
We should validate a bit more than the name to make sure we aren't renaming someone's custom implementation. Best to verify the model vs table structure.

Reply:
Hey Derrick, yes that's a definitely a concern we need to have, but but do you think that can still occur here? The `hasTable` here is just an extra check, but when defining `shouldMigrate` we're checking if the content type `admin::audit-logs` has had its collection name updated from `audit_logs` to `strapi_audit_logs`. With the `admin::` prefix I don't think it would impact any user's content type

Reply:
Unless they used patch package to inject the content-type. Better safe than sorry to ensure the attributes of the collection type match exactly to the DB tables. This is something @gu-stav mentioned in his comment above too.

Reply:
Got it, I added another check